internal/internal_worker.go

```diff
@@ -867,6 +938,17 @@ func validateFnFormat(fnType reflect.Type, isWorkflow bool) error {
 		}
 	}
 
+	if isDynamic {
+		if fnType.NumIn() != 2 {
+			return fmt.Errorf(
+				"expected function to have two arguments, first being workflow.Context and second being an EncodedValues type, found %d arguments", fnType.NumIn(),
+			)
+		}
+		if !fnType.In(1).Implements(reflect.TypeOf((*converter.EncodedValues)(nil)).Elem()) {
```

Comment:
```suggestion
		if fnType.In(1) != reflect.TypeOf((*converter.EncodedValues)(nil)).Elem() {
```

I think simple equality check is good here